napari/_qt/updates.py

```diff
@@ -0,0 +1,225 @@
+import sys
+import uuid
+from pathlib import Path
+from time import sleep
+
+from qtpy.QtCore import QObject, QProcess, Signal
+
+from ..utils.notifications import notification_manager
+from ..utils.translations import trans
+from .qthreading import create_worker
+
+
+class UpdateManager(QObject):
```

Comment:
This is very similar to the `Installer` handling plugin installation (for the conda side) I want to eventually unify them.